File: task-03/Javascript/Sub task 4/subtask4.js
const fs = require('fs');

// Function to generate diamond pattern
function printDiamond(n) {
    if (n < 1) {
        return "Number should be greater than 1\n";
    }

    let result = '';
    const mid = Math.floor(n / 2);

    // Upper part of the diamond
    for (let i = 1; i <= n; i += 2) {
        const spaces = mid - Math.floor(i / 2);
        result += ' '.repeat(spaces) + '*'.repeat(i) + '\n';
    }

    // Lower part of the diamond
    if (n % 2 === 0) {
        for (let i = n - 1; i >= 1; i -= 2) {
            const spaces = mid - Math.floor(i / 2) + 1;
            result += ' '.repeat(spaces - 1) + '*'.repeat(i) + '\n';
        }
    } else {
        for (let i = n - 2; i >= 1; i -= 2) {
            const spaces = mid - Math.floor(i / 2);
            result += ' '.repeat(spaces) + '*'.repeat(i) + '\n';
        }
    }

    return result;
}

// Main function
function main() {
    fs.readFile('input.txt', 'utf8', (err, data) => {
        if (err) {
            console.error("Failed to open input.txt:", err);
            return;
        }

        const n = parseInt(data.trim(), 10);
        if (isNaN(n)) {
            console.error("Invalid number in input.txt");
            return;
        }

        const diamond = printDiamond(n);

        fs.writeFile('output.txt', diamond, (err) => {
            if (err) {
                console.error("Failed to write to output.txt:", err);
                return;
            }
            console.log("File transfer was successful");
        });
    });
}

main();

File: task-03/Java/Sub task 4/subtask4.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;

public class subtask4 {

    public static void printDiamond(BufferedWriter output, int n) throws IOException {
        if (n < 1) {
            output.write("Number should be greater than 1\n");
            return;
        }

        int mid = n / 2;

        // Upper part of the diamond
        for (int i = 1; i <= n; i += 2) {
            int spaces = mid - i / 2;
            output.write(" ".repeat(spaces) + "*".repeat(i) + "\n");
        }

        // Lower part of the diamond
        if (n % 2 == 0) {
            for (int i = n - 1; i >= 1; i -= 2) {
                int spaces = mid - i / 2 + 1;
                output.write(" ".repeat(spaces - 1) + "*".repeat(i) + "\n");
            }
        } else {
            for (int i = n - 2; i >= 1; i -= 2) {
                int spaces = mid - i / 2;
                output.write(" ".repeat(spaces) + "*".repeat(i) + "\n");
            }
        }
    }

    public static void main(String[] args) {
        try (BufferedReader input = new BufferedReader(new FileReader("input.txt"));
             BufferedWriter output = new BufferedWriter(new FileWriter("output.txt"))) {

            String line = input.readLine();
            if (line != null) {
                int n = Integer.parseInt(line.trim());
                printDiamond(output, n);
                System.out.println("File transfer was successful");
            } else {
                System.out.println("Input file is empty.");
            }

        } catch (IOException e) {
            System.err.println("An error occurred: " + e.getMessage());
        }
    }
}


